app/feature/feature-claim-details/src/main/kotlin/com/hedvig/android/feature/claim/details/ui/ClaimDetailsDestination.kt

```diff
@@ -104,46 +126,100 @@ private fun ClaimDetailScreen(
 private fun ClaimDetailScreen(
   uiState: ClaimDetailUiState.Content,
   onChatClick: () -> Unit,
+  openUrl: (String) -> Unit,
+  imageLoader: ImageLoader,
   modifier: Modifier = Modifier,
 ) {
-  Column(
+  LazyVerticalGrid(
+    columns = GridCells.Fixed(3),
+    verticalArrangement = Arrangement.spacedBy(8.dp),
+    horizontalArrangement = Arrangement.spacedBy(8.dp),
     modifier = modifier
-      .verticalScroll(rememberScrollState())
       .windowInsetsPadding(WindowInsets.safeDrawing.only(WindowInsetsSides.Horizontal + WindowInsetsSides.Bottom))
       .padding(horizontal = 16.dp),
   ) {
-    Spacer(Modifier.height(8.dp))
-    ClaimStatusCard(
-      uiState = uiState.claimStatusCardUiState,
-      onClick = null,
-    )
-    Spacer(Modifier.height(8.dp))
-    ClaimInfoCard(uiState.claimStatus, uiState.claimOutcome, onChatClick)
-    when (uiState.submittedContent) {
-      is ClaimDetailUiState.Content.SubmittedContent.Audio -> {
-        Spacer(Modifier.height(40.dp))
-        ClaimDetailHedvigAudioPlayerItem(uiState.submittedContent.signedAudioURL)
-      }
-      is ClaimDetailUiState.Content.SubmittedContent.FreeText -> {
+    item(span = { GridItemSpan(3) }) {
+      Column {
+        Spacer(Modifier.height(8.dp))
+        ClaimStatusCard(
+          uiState = uiState.claimStatusCardUiState,
+          onClick = null,
+        )
+        Spacer(Modifier.height(8.dp))
+        ClaimInfoCard(uiState.claimStatus, uiState.claimOutcome, onChatClick)
         Spacer(Modifier.height(40.dp))
         Text(
-          stringResource(R.string.claim_status_detail_submitted_message),
+          stringResource(R.string.claim_status_detail_uploaded_files_info_title),
           Modifier.padding(horizontal = 2.dp),
         )
         Spacer(Modifier.height(8.dp))
-        HedvigCard(Modifier.fillMaxWidth()) {
-          Text(
-            uiState.submittedContent.text,
-            Modifier.padding(16.dp),
+        when (uiState.submittedContent) {
+          is ClaimDetailUiState.Content.SubmittedContent.Audio -> {
+            ClaimDetailHedvigAudioPlayerItem(uiState.submittedContent.signedAudioURL)
+          }
+
+          is ClaimDetailUiState.Content.SubmittedContent.FreeText -> {
+            HedvigCard(Modifier.fillMaxWidth()) {
+              Text(
+                uiState.submittedContent.text,
+                Modifier.padding(16.dp),
+              )
+            }
+          }
+
+          else -> {}
+        }
+        Spacer(Modifier.height(16.dp))
+      }
+    }
+
+    items(uiState.files) {
+      Box(
+        Modifier
+          .background(
+            shape = MaterialTheme.shapes.squircleMedium,
+            color = MaterialTheme.colorScheme.surface,
           )
+          .clickable {
+            openUrl(it.url)
+          },
+        contentAlignment = Alignment.Center,
+      ) {
+        if (it.mimeType.contains("image")) {
+          ClaimAsyncImage(
```

Comment:
You'll probably want to pass in a cacheKey too, if the file contains some sort of unique ID that will be perfect.
Because otherwise each time from the backend we get a new URL to the image itself, and we end up having coil save into cache all those images (of the same image) in the cache, and we also never end up loading it quickly without doing another network request.